[core][release] many_tasks failed with not using all cpus

### What happened + What you expected to happen

[Recent fail run](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_1uzVPB2drY1KzaFEdBJnU9CH)

```
(MemoryMonitorActor pid=1566) 214       0.02GiB /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet --raylet_socket_name=
Traceback (most recent call last):
  File "distributed/test_many_tasks.py", line 107, in <module>
    test()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "distributed/test_many_tasks.py", line 70, in test
    test_max_running_tasks(num_tasks)
  File "distributed/test_many_tasks.py", line 44, in test_max_running_tasks
    assert max_cpus - min_cpus_available > threshold, err_str
AssertionError: Only 879.25/4096.0 cpus used.
```

### Versions / Dependencies

Has been flaky rerun usually succeeds. 
First failure [58255c7](https://github.com/ray-project/ray/commit/58255c79878b724288a523bf99c97e4257040c3c)
Last success [6486127](https://github.com/ray-project/ray/commit/64861270b8fceadd495229afb297c5e36351ebb5)


58255c798 Widening range of grpcio versions allowed. (#28623)
64861270b [Datasets] Use generators for block splitting (#28682)

### Reproduction script

NA

### Issue Severity

_No response_

--- a/release/benchmarks/distributed/test_many_tasks.py
+++ b/release/benchmarks/distributed/test_many_tasks.py
@@ -6,7 +6,7 @@ import ray._private.test_utils as test_utils
 import time
 import tqdm
 
-sleep_time = 150
+sleep_time = 300
 
 
 def test_max_running_tasks(num_tasks):

--- a/release/benchmarks/many_nodes.yaml
+++ b/release/benchmarks/many_nodes.yaml
@@ -14,7 +14,7 @@ head_node_type:
 
 worker_node_types:
     - name: small_worker
-      instance_type: m5.xlarge
+      instance_type: m5.2xlarge
       min_workers: 249
       max_workers: 249
       use_spot: false

